format/midi/midievents.go

```diff
@@ -0,0 +1,144 @@
+package midi
+
+import (
+	"github.com/wader/fq/pkg/decode"
+	"github.com/wader/fq/pkg/scalar"
+)
+
+type MidiEventType byte
+
+const (
+	TypeNoteOff            MidiEventType = 0x80
+	TypeNoteOn             MidiEventType = 0x90
+	TypePolyphonicPressure MidiEventType = 0xa0
+	TypeController         MidiEventType = 0xb0
+	TypeProgramChange      MidiEventType = 0xc0
+	TypeChannelPressure    MidiEventType = 0xd0
+	TypePitchBend          MidiEventType = 0xe0
+)
+
+var midievents = scalar.UintMapSymStr{
```

Comment:
When using `*MapSymStr` i usually try to make the mapped string jq-identifer-ish, so ex `snake_case`, but is not required if some specification etc uses some other naming or it's confusing

Reply:
I back and forth'ed on that so many times :-). I've reworked it so that the _event types_ and _event names_ are snake-cased but left the informational fields as is, e.g.
```
[13]{}: sequencer_specific_event 0xde-0xe8 (10)
  time{}: 0xde-0xdf (1)
    delta: 0 0xde-0xdf (1)
    tick: 0
  event: "sequencer_specific_event" (65407) 0xdf-0xe1 (2)
  info{}: 0xe1-0xe8 (7)
    bytes: "[0 0 59 58 76 94]" 0xe1-0xe8 (7)
    manufacturer: "Mark Of The Unicorn (MOTU)" ("003B")
    data: "[58 76 94]"
```

Let me know what you think..

Reply:
Hehe i know the feeling, hard to decide how to modell things. I will probably have to build and playaround a bit to get a feeling about this

Reply:
Added some comments about this, see comment in .fqtest files